lib/email/receiver.rb

```diff
@@ -660,102 +660,97 @@ def trim_discourse_markers(reply)
       reply.split(reply_above_line_regex)[0]
     end
 
-    def parse_from_field(mail = nil)
-      mail ||= @mail
-
+    def parse_from_field(mail, process_forwarded_emails: true)
```

Comment:
Added the `process_forwarded_emails` argument to ensure we don't process "embedded emails" on already embedded emails.